test/framework/cluster.go

```diff
@@ -675,6 +682,9 @@ func (e *ClusterE2ETest) Run(name string, args ...string) {
 		}
 
 		if errorMessage != "" {
+			if e.ExpectFailure {
+				return
```

Comment:
Can we add a print statement here saying that we expected failure and some context?